src/frontend/parser.mly

```diff
@@ -218,61 +230,108 @@ unsized_dims:
   | LBRACK cs=list(COMMA) RBRACK
     { grammar_logger "unsized_dims" ; List.length(cs) }
 
-(* declarations *)
+(* Never accept this rule, but return the same type as expression *)
+no_assign:
+  | UNREACHABLE
+    { (* This code will never be reached *)
```

Comment:
Is there any time when this could be executed? I didn't see anything in the lexer that generates this token. If there's nothing that could cause it to be executed, might be nicer to just raise here instead.

Reply:
Yeah good point, this should raise an error.